### ISSUE TITLE
remove_iov error

### ISSUE THREAD

**Issue by SCarter79** [pharmpy/pharmpy#538]
R: 4.1.2, Python 3.9.6, pharmr and pharmpy: 0.49.0
I was trying to copy a model, remove_iov on a quite complex IOV model then save the model to a new .mod file (original .mod file at bottom):

```
mod <- read_model(here('runs/FactorVIII/model.mod'))
print_model_code(mod)

mod1 <- copy_model(mod) %>%
  remove_iov() %>%
  write_model(here('runs/FactorVIII/model_0IOV.mod'), force = TRUE) 

print_model_code(mod1)
```

but got the following error:

```
> py_last_error()
$type
[1] "IndexError"

$value
[1] "list index out of range"

$traceback
[1] "  File \"C:\\Users\\AppData\\Local\\R-MINI~1\\envs\\R-RETI~1\\lib\\site-packages\\pharmpy\\modeling\\common.py\", line 166, in write_model\n    model.update_source(path=path, force=force)\n"                                   
[2] "  File \"C:\\Users\\AppData\\Local\\R-MINI~1\\envs\\R-RETI~1\\lib\\site-packages\\pharmpy\\plugins\\nonmem\\model.py\", line 166, in update_source\n    update_statements(self, self._old_statements, self._statements, trans)\n"
[3] "  File \"C:\\Users\\AppData\\Local\\R-MINI~1\\envs\\R-RETI~1\\lib\\site-packages\\pharmpy\\plugins\\nonmem\\update.py\", line 450, in update_statements\n    rec.statements = main_statements\n"                                 
[4] "  File \"C:\\Users\\AppData\\Local\\R-MINI~1\\envs\\R-RETI~1\\lib\\site-packages\\pharmpy\\plugins\\nonmem\\records\\code_record.py\", line 486, in statements\n    kept.append(self.root.children[node_index])\n"               

$message
[1] "IndexError: list index out of range"
```

model.mod:
```
;Basal FVIII model Clinical data ignored
;PREP (grouped) as covariate on V1
;PREP (grouped) as covariate on CL SAME THETA
;age as covariate on CL only

$PROBLEM 2-comp, allometric scaling, IOV on CL and V
$INPUT ID DAT2=DROP OTIM=DROP TIME DV AMT RATE AGE WT
BLGR PREP EVID OCC YBIN OCC2 STUD
$DATA DataFVIII6.prn IGNORE=(STUD.EQ.1) IGN=@ 
$SUBROUTINES ADVAN3 TRANS4

$PK
Q1 = 0
Q2 = 0
Q3 = 0
Q4 = 0
Q5 = 0
Q6 = 0
Q7 = 0
Q8 = 0
IF(OCC.EQ.1)Q1 = 1
IF(OCC.EQ.2)Q2 = 1
IF(OCC.EQ.3)Q3 = 1
IF(OCC.EQ.4)Q4 = 1
IF(OCC.EQ.5)Q5 = 1
IF(OCC.EQ.6)Q6 = 1
IF(OCC.EQ.7)Q7 = 1
IF(OCC.EQ.8)Q8 = 1

IF (PREP.EQ.1) PREP2 = 1
IF (PREP.EQ.2) PREP2 = 2
IF (PREP.EQ.3) PREP2 = 1
IF (PREP.EQ.4) PREP2 = 1
IF (PREP.EQ.5) PREP2 = 1
IF (PREP.EQ.6) PREP2 = 2
IF (PREP.EQ.7) PREP2 = 2
;IF (PREP.EQ.8) PREP2 = 3
IF (PREP.EQ.9) PREP2 = 2

IF (PREP2.EQ.1) THEN
TVCL = THETA(1)*((WT/68)**THETA(7))*(1+THETA(13)*(AGE-24)) ;68 = Median of body weight
TVV1 = THETA(2)*((WT/68)**THETA(8)) 
ENDIF
IF (PREP2.EQ.2) THEN
TVCL = THETA(1)*((WT/68)**THETA(7))*(1+THETA(12))*(1+THETA(13)*(AGE-24))
TVV1 = THETA(2)*((WT/68)**THETA(8))*(1+THETA(12))
ENDIF
TVQ  = THETA(5)*(WT/68)**THETA(9)
TVV2 = THETA(6)*(WT/68)**THETA(10)
IOVCL= Q1*ETA(4)+Q2*ETA(5)+Q3*ETA(6)+Q4*ETA(7)+Q5*ETA(8)
IOVCL2 = IOVCL+Q6*ETA(9)+Q7*ETA(10)+Q8*ETA(11)
IOVV = Q1*ETA(12)+Q2*ETA(13)+Q3*ETA(14)+Q4*ETA(15)
IOVV2= IOVV+Q5*ETA(16)+Q6*ETA(17)+Q7*ETA(18)+Q8*ETA(19)
CL   = TVCL*EXP(ETA(2)+IOVCL2)
V1   = TVV1*EXP(ETA(3)+IOVV2)
Q    = TVQ
V2   = TVV2

TVBA = THETA(11)
BASE = TVBA*EXP(ETA(1))

S1 = V1

K = CL/V1
K12 = Q/V1
K21 = Q/V2

$ERROR
IPRED = F + BASE
IRES  = DV-IPRED
W     = SQRT(THETA(3)**2 + (THETA(4)*IPRED)**2)
IWRES = IRES/W
;IF(W.EQ.0) W = 1

Y = IPRED + W*EPS(1)

$THETA             
(0,250)            ;1 CL (ml/h)
(0,3380)           ;2 V1 (ml)
(0,0.01)           ;3 Add res error (SD, units = U/ml)
(0,0.08)           ;4 Prop res error (Rel SD = CV)
(0,444)            ;5 Q (ml/h)
(0,268)            ;6 V2 (ml)
0.75 FIX           ;7 Scaling for CL
1 FIX              ;8 Scaling for V1
0.75 FIX           ;9 Scaling for Q
1 FIX              ;10 Scaling for V2
(0,0.01)           ;11 Baseline
(-0.2)             ;12 Difference in TVCL and TVV1 for full-lenght recombinant FVIII
(-0.0185,-0.0076,0.055) ;13 Fractional change in CL per year different from 24 years

$OMEGA BLOCK(3)
0.141                 ;varBase
0.0611 0.107          ;covarBase-CL varETACL
0 0.0509 0.0397       ;covar Base-V1 covar CL-V1 varV1


$OMEGA BLOCK(1) 0.0381 ;IOV for CL
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) 0.0410 ;IOV for V
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME
$OMEGA BLOCK(1) SAME

$SIGMA 1 FIX    
$ESTIMATION METHOD=1 INTER PRINT=1 MAXEVAL=9999
$COVARIANCE PRECOND=1 PFCOND=1

$TABLE ID TIME IPRED CWRES
      NOPRINT ONEHEADER FILE=sdtab249
```

**Comment by ZheHuang96** [comment]
Hi Simon,

May I ask a naive question? Is it the same error as the error you just mentioned? I compared them for a while, the error is `IndexError: list index out of range ` and I guess you use the same model file and use `remove_iov()` on it?

Thank you!

Best wishes,
Zhe Huang

**Comment by ZheHuang96** [comment]
Ok, I see! Thank you for your reply!

**Comment by ZheHuang96** [comment]
I think this issue can be closed now? @pharmpy-dev-123  There are some other issues related with iov. Maybe you can have a quick check and close them :)

**Comment by pharmpy-dev-123** [comment]
Should have been fixed by #771 I hope. Will add a regression test to make sure. I'll check for other IOV-related issues.
